### PR TITLE
Use timeout from the HTTP client

### PR DESCRIPTION
**Pull request by lucacome**
Make timeout customizable.

### CODE REVIEW

**Comment by pleshakov** [comment]
@lucacome 
Would it make sense to extend the API so it supports passing the ctx from the caller? For example, have a method `GetStatsWithContext(ctx *context.Context)`.

This way, the caller will be able to cancel the request - either because of the timeout or explicitly canceling using the context cancel function. 

At the same time, the timeout setting of the http.Client will remain independent from the context. And the existing methods will remain backward compatible. We can change them so that they call the new methods using `context.TODO()`

**Review by jjngx — APPROVED** [review]
👍🏻

**Review by tomasohaodha — APPROVED** [review]

File: client/nginx.go [hunk 1]
```diff
@@ -535,7 +534,7 @@ func versionSupported(n int) bool {
 }
 
 func getAPIVersions(httpClient *http.Client, endpoint string) (*versions, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), httpClient.Timeout)
```
Comment (tomasohaodha) [inline client/nginx.go:537]: How many downstream projects use nginx-plus-go-client? Can we be sure this change is non-breaking for them?

Comment (jjngx) [reply client/nginx.go:537]: Default [Go http client do not specify Timeout](https://github.com/golang/go/blob/213504e5431891e8421a92ab72ac08981ce07516/src/net/http/client.go#L91L107) (means no timeout), therefore it should not be used in a prod env. Whoever uses this library needs to pass his own http client and remember to explicitly setup Timeout.


One option to reduce required "paperwork" from a user of the lib is to provide a min default http client with specified timeout, and ability to use a func option if the user wants to configure NGINX plus client with a custom http client.

**Pull request closed without merging**